src/main/java/games/strategy/engine/GameEngineUtils.java

```diff
@@ -0,0 +1,52 @@
+package games.strategy.engine;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Comparator;
+
+import games.strategy.util.Version;
+
+/**
+ * A collection of methods for use with game engines.
+ */
+public final class GameEngineUtils {
+  private GameEngineUtils() {}
+
+  /**
+   * Indicates the engines with the specified versions are compatible.
+   *
+   * @param engineVersion The first engine version.
+   * @param otherEngineVersion The second engine version.
+   *
+   * @return {@code true} if the engines are compatible; otherwise {@code false}.
+   */
+  public static boolean isEngineCompatibleWithEngine(final Version engineVersion, final Version otherEngineVersion) {
+    checkNotNull(engineVersion);
+    checkNotNull(otherEngineVersion);
+
+    return compareVersionsExcludingMicro(engineVersion, otherEngineVersion) == 0;
+  }
+
+  private static int compareVersionsExcludingMicro(final Version version, final Version otherVersion) {
+    return Comparator.comparingInt(Version::getMajor)
+        .thenComparingInt(Version::getMinor)
+        .thenComparingInt(Version::getPoint)
+        .compare(version, otherVersion);
+  }
+
+  /**
+   * Indicates the engine with the specified version is compatible with a map having the specified minimum engine
+   * version.
+   *
+   * @param engineVersion The engine version.
+   * @param mapMinimumEngineVersion The minimum engine version required by the map.
+   *
+   * @return {@code true} if the engine is compatible with the map; otherwise {@code false}.
+   */
+  public static boolean isEngineCompatibleWithMap(final Version engineVersion, final Version mapMinimumEngineVersion) {
```

Comment:
Agreed about the readability.  However, I still question if we should have _engine_-specific version methods in the _general_ `Version` class.

What if, instead of the static methods in `GameEngineUtils`, we had a kind of throwaway wrapper class for game engine versions to host these methods?  Something like:

```java
Version version = // ...
Version otherVersion = // ...

GameEngineVersion.of(version).isCompatibleWithEngine(otherVersion)
GameEngineVersion.of(version).isCompatibleWithMap(otherVersion)
```

Reply:
I'd just as well attach those methods to `Version.java` , but I've no strong objection to attaching to a wrapper `GameEngineVersion`. The biggest key for me was readability and notably converting from static methods to instance.

Possibly a side note, but in case it is relevant: MapVersion is interesting and we may want to keep in mind the future plans. The current map version spec I don't think is that intuitive, nor does it allow for a good degree of backward compatible changes. I'd like for us having a parser strategy to replace the current map XML parser that would allow us to have multiple parsers that could function across multiple different file specs.